lib/client/mfa.go

```diff
@@ -79,5 +82,22 @@ func (tc *TeleportClient) newPromptConfig(opts ...mfa.PromptOpt) *libmfa.PromptC
 		cfg.WebauthnLoginFunc = tc.WebauthnLogin
 		cfg.WebauthnSupported = true
 	}
+
 	return cfg
 }
+
+// NewSSOMFACeremony creates a new SSO MFA ceremony.
+func (tc *TeleportClient) NewSSOMFACeremony(ctx context.Context) (mfa.SSOMFACeremony, error) {
+	rdConfig, err := tc.ssoRedirectorConfig(ctx, "" /*connectorDisplayName*/)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	rd, err := sso.NewRedirector(rdConfig)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	context.AfterFunc(ctx, rd.Close)
```

Comment:
Let's document that this happens, as typically I would not expect the context to have an effect after the method returns.

I would usually favor a design where ownership is clearer - so maybe the SSOMFACeremony exposes a Close method and the caller has to defer-call it (or AfterFunc or whatever they prefer). WDYT?